api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java

```diff
@@ -52,22 +50,26 @@ public class ObsBasedDiagnosticReportService extends BaseFhirService<DiagnosticR
 	
 	@Autowired
 	private DiagnosticReportObsValidator diagnosticReportObsValidator;
-
+	
 	@Autowired
 	private DiagnosticReportRequestValidator diagnosticReportRequestValidator;
 	
+	@Autowired
+	private OrderUpdateService orderUpdateService;
+	
 	@Autowired
 	private SearchQuery<FhirDiagnosticReport, DiagnosticReport, FhirDiagnosticReportDao, ObsBasedDiagnosticReportTranslator, SearchQueryInclude<DiagnosticReport>> searchQuery;
 	
 	@Autowired
 	private SearchQueryInclude<DiagnosticReport> searchQueryInclude;
+	
 	@Override
 	public DiagnosticReport create(@Nonnull DiagnosticReport diagnosticReport) {
 		try {
 			FhirDiagnosticReport fhirDiagnosticReport = obsBasedDiagnosticReportTranslator.toOpenmrsType(diagnosticReport);
 			diagnosticReportObsValidator.validate(fhirDiagnosticReport);
-			if (diagnosticReport.getBasedOn().size() > 0 && diagnosticReport.getBasedOn().get(0).getReference() != null)
-				diagnosticReportRequestValidator.validate(fhirDiagnosticReport);
+			diagnosticReportRequestValidator.validate(diagnosticReport, fhirDiagnosticReport);
+			orderUpdateService.updateOrder(diagnosticReport, fhirDiagnosticReport);
```

Comment:
Should we update the order after saving the diagnostic report. As in I am thinking of the scenario where after updating the order, there is some exception while creating/updating the diagnostic report.

Reply:
Moved order update logic after creating/updating diagnostic report